### DISCUSSION TITLE
Steam Directory

### DISCUSSION THREAD

**Discussion by justxanderscode** [ThioJoe/Windows-Super-God-Mode#22] [Ideas & Suggestions]
Please add a Feature which skipps the Steam directory so the Game files will be skipped...

Think of People with a 4 TB SSD and its fully packed with Games... thats not fun :(

But as for now, good work!

**Comment by ThioJoe** [comment]
Which version of the script are you using? Version 1.2.0 should already do this

**Reply by justxanderscode** [reply]
Hey ThioJoe,

then it should probably be because I have my Steam games on the D:\ drive ☠️ (to be exact, the whole Steam installation is on the D:\ drive)

I hope this doesn't mean that you have to include every drive letter...

Had it run from like 22 pm local time to 6 am local time and it only was at 52 %. Have a 2 TB SSD. 🥲😂

**Reply by ThioJoe** [reply]
What's the name of the folder with the steam games? I think I was going to add SteamLibrary as a value that would limit the searching but might have forgot or decided against it for some reason.

You can try editing this line (line 3480 on v1.2.1):
`$dontSearchFolders = @($env:WINDIR)`

To say this for example (or whatever the library folder name is called). Don't forget the wildcard asterisk * at the beginning:
`$dontSearchFolders = @($env:WINDIR, '*SteamLibrary')`

Or alternatively you could be more specific and put the whole path to the folder between the quotes like:
`$dontSearchFolders = @($env:WINDIR, 'D:\SteamLibrary')`

And that should do the trick. You can add as many paths to the list like that as you need. For any programs it goes to search where it detects a matching folder path matching that, it will only search the single binary file instead of the entire directory.

-----

On a related note:

If it still takes forever you can enable debug logging mode and it should produce a text file that lists every single file that will be searched so you can maybe scroll through and see if there are any directories that have an enormous amount of files slowing it down.

There's another variable called `$ignorePaths` on this line below which is subtly different, which you can set to ignore certain specific folders within a larger program installation directory, but still allow the rest of the the directory to be searched:
`$ignorePaths = @('\steamapps\' )`

By default it just has steamapps as a value like, that meaning the steam installation directory (`C:\Program Files (x86)\Steam`) will be searched when looking for the `steam://` protocol, but within that it will ignore the `steamapps` folder and anything in it. Note that the patterns for that variable are regex not wildcards so it's a bit different. But for simple folder name match you can just put the exact name with the backslashes around it. I believe it's case sensitive.

You can also add multiple more values if you want like:
`$ignorePaths = @('\steamapps\', '\AFileNameYouWantToIgnore.exe',  '\WhateverFolderNameToIgnore')`

Though I'll have to double check if I have that implemented right, I might need to set that to escape the regex patterns or it might not be working as expected 🤔

**Reply by ThioJoe** [reply]
Ok actually it turns out the patterns weren't matching correctly so \steamapps\ would have been being searched when it shouldn't have. I just fixed it in the new 1.2.2 release: https://github.com/ThioJoe/Windows-Super-God-Mode/releases/tag/v1.2.2

The stuff I wrote above still applies, though in the script I renamed and reworked how some of the variables I mentioned above work a bit to be more clear. `$ignorePaths` is the same but I renamed I renamed `$dontSearchFolders` to now be called `$foldersToOnlySearchBinary`.

Also now for `$foldersToOnlySearchBinary`, all the wildcards are just in the variable when declared instead of some being added later, which just means don't forget to include the wildcard at the end whereas before you didn't have to put it there yourself. You can see how it looks by default for example:
`$foldersToOnlySearchBinary = @("$env:WINDIR\*", "*\SteamLibrary\*")`

----

**_Anyway_** the takeaway for you again is that if you still notice it searching game folders you want to ignore because your steam library location is called something other than `SteamLibrary`, you can add it to the line like:
`$foldersToOnlySearchBinary = @("$env:WINDIR\*", "*\SteamLibrary\*", "*\WhateverSteamLibraryName\*")`

You shouldn't really have to bother with $ignorePaths though again you can if you want it to only skip a certain sub-folder but keep searching the rest of a directory.
